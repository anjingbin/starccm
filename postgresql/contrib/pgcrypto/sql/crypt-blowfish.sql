--
-- crypt() and gen_salt(): bcrypt
--

select crypt('', '$2a$06$RQiOJ.3ELirrXwxIZY8q0O');

select crypt('foox', '$2a$06$RQiOJ.3ELirrXwxIZY8q0O');

create table ctest (data text, res text, salt text);
insert into ctest values ('password', '', '');

update ctest set salt = gen_salt('bf', 8);
update ctest set res = crypt(data, salt);
select res = crypt(data, res) as "worked" from ctest;

drop table ctest;

