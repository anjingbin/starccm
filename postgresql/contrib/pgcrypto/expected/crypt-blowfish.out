--
-- crypt() and gen_salt(): bcrypt
--
select crypt('', '$2a$06$RQiOJ.3ELirrXwxIZY8q0O');
                            crypt                             
--------------------------------------------------------------
 $2a$06$RQiOJ.3ELirrXwxIZY8q0OlGbBEpDmx7IRZlNYvGJ1SHXwNi2cEKK
(1 row)

select crypt('foox', '$2a$06$RQiOJ.3ELirrXwxIZY8q0O');
                            crypt                             
--------------------------------------------------------------
 $2a$06$RQiOJ.3ELirrXwxIZY8q0OR3CVJrAfda1z26CCHPnB6mmVZD8p0/C
(1 row)

create table ctest (data text, res text, salt text);
insert into ctest values ('password', '', '');
update ctest set salt = gen_salt('bf', 8);
update ctest set res = crypt(data, salt);
select res = crypt(data, res) as "worked" from ctest;
 worked 
--------
 t
(1 row)

drop table ctest;
